utils/update_copyright.py

```diff
@@ -17,39 +30,52 @@ def main():
         # Fix copyright headers in C++ files
+        regex_header = re.compile(
+            '(.*Copyright \(C\) \d\d\d\d)(.*)( by the Widelands Development Team.*)')
+
+        def repl_header(match):
+            if match.group(1).endswith(year):
+                return match.group(1) + match.group(3)
+            else:
+                return match.group(1) + '-' + year + match.group(3)
         for filename in find_files(src_path, ['.h', '.cc']):
-            sys.stdout.write('.')
-            sys.stdout.flush()
-            lines = read_text_file(filename).strip().split('\n')
-            new_lines = []
-            regex = re.compile(
-                '(.*Copyright \(C\) \d\d\d\d)(.*)( by the Widelands Development Team.*)')
-            for line in lines:
-                match = regex.match(line)
-                if match:
-                    if match.group(1).endswith(year):
-                        line = match.group(1) + match.group(3)
-                    else:
-                        line = match.group(1) + '-' + year + match.group(3)
-                new_lines.append(line.rstrip() + '\n')
-            write_text_file(filename, ''.join(new_lines))
+            update_file(filename, regex_header, repl_header)
 
         # Now update the Buildinfo
         filename = os.path.join(src_path, 'build_info.h')
-        lines = read_text_file(filename).strip().split('\n')
-        new_lines = []
         regex = re.compile(
             '(.*constexpr uint16_t kWidelandsCopyrightEnd = )(\d\d\d\d)(;)')
-        for line in lines:
-            match = regex.match(line)
-            if match:
-                line = match.group(1) + year + match.group(3)
-            new_lines.append(line.rstrip() + '\n')
-        write_text_file(filename, ''.join(new_lines))
+
+        def replace(match): return match.group(1) + year + match.group(3)
+        update_file(filename, regex, replace)
+
+        # Now update special files
+        filename = os.path.join(base_path, 'data/txts/LICENSE.lua')
+        regex = re.compile(
+            '(.*\"Copyright 2002 - %1% by the Widelands Development Team\.\"\):bformat\()(\d\d\d\d)(\).*)')
+        update_file(filename, regex, replace)
+
+        filename = os.path.join(
+            base_path, 'utils/win32/innosetup/Widelands.iss')
+        regex = re.compile(
+            '(#define Copyright \"Widelands Development Team 2001\-)(\d\d\d\d)(\")')
+        update_file(filename, regex, replace)
+
+        filename = os.path.join(base_path, 'utils/win32/widelands.rc.cmake')
+        update_file(filename, regex_header, repl_header)
+
+        filename = os.path.join(base_path, 'debian/copyright')
+        regex = re.compile(
+            '(Copyright: 2002\-)(\d\d\d\d)( by the Widelands Development Team)')
+        update_file(filename, regex, replace)
+
         print(' done.')
+        print(
+            f'You might need to update some copyright years in {filename} manually.')
```

Comment:
f-strings are only available since [python version 3.6](https://docs.python.org/3/reference/lexical_analysis.html#f-strings). Some developers are using Ubuntu or Debian, which ships python 2.7 by default.

Reply:
Thanks for the hint, I changed it to % formatting like in other scripts.